src/Generation/GapicClientV2Generator.php

```diff
@@ -285,10 +286,12 @@ private function operationMethods(): Vector
         if (!$this->serviceDetails->hasLro && !$this->serviceDetails->hasCustomOp) {
             return Vector::new([]);
         }
-
-        $ctype = $this->serviceDetails->hasCustomOp ?
-            $this->serviceDetails->customOperationServiceClientType :
-            Type::fromName(OperationsClient::class);
+        $ctype = $this->serviceDetails->hasCustomOp
+            ? $this->serviceDetails->customOperationServiceClientType
+            : Type::fromName($this->serviceDetails->migrationMode === MigrationMode::NEW_SURFACE_ONLY
```

Comment:
Should we do this for `MIGRATING` too?

Reply:
My reasoning here was that, since we've already "Gone GA" with the new surfaces, we need to wait until they're behind a major version (if the library itself is already stable) to make this technically breaking change.

If a library is `NEW_SURFACE_ONLY`, it means either that (1) the library is v2, OR that the library NEVER had a V1 surface and is pre-GA (1.0). In both these scenarios, we can push this "breaking change".  However, for libraries that are "Migrating", they could be 1.0+, and therefore would break with this change.

It's not perfect because we do have some pre-GA libraries which are in "MIGRATING" mode that we COULD push this breaking change out to, but won't be able to until we bump to "NEW_SURFACE_ONLY". I don't really have a good suggestion for that.